Document cache handling mechanism and improve patch method documentation (#1595)
PrestaShop/docs: webservice/getting-started.md

The `sendemail=1` parameter can be used on the `order_carriers` endpoint to send the _in-transit_ email with the tracking number. Example: `http://example.com/api/order_carriers/12345?sendemail=1`
12345 is the order carrier id.

##### Cache handling

> A cache mechanism has been introduced and bug fixed in {{< minver v="8.0" >}}, it allows to detect if the content changed or not between your previous API call and the current one. 

Reviewer: ```suggestion
A cache mechanism has been introduced and bug fixed in {{< minver v="8.0" >}}, it allows you to detect if the content changed between your API calls. 
```

To use it:

1) in your first API call, retrieve the header `Content-Sha1` and store it on your side.
2) in your second API call, add a `Local-Content-Sha1` header, with the previously stored `Content-Sha1` value. 

If the content has not changed, the API will return a `304 Not Modified` response code.
If the content has changed, the API will return a `200 Ok` response code. 

{{% notice note %}}

> It can be used to avoid un-necessary updates if the resource didn't changed since last API call. 

Reviewer: ```suggestion
It can be used to avoid unnecessary updates if the resource didn't change since the last API call. 
```
